src/Npgsql/NpgsqlDatabaseInfo.cs

```diff
@@ -204,6 +204,44 @@ protected static Version ParseServerVersion(string value)
             return new Version(versionString);
         }
 
+        /// <summary>
+        /// Allocation free helper function to find if version is greater than expected
+        /// </summary>
+        private static bool VersionGreaterOrEqual(Version version, int major, int minor, int build)
```

Comment:
Some nits:
* Let's make this an extension method over Version (`IsGreaterOrEqual`) - please move this to VersionExtensions.cs under the Util folder (non-public class).
* This method can benefit from lots of use of the conditional operator to make it terser.
* We don't specify `private` as it's the default.
* We don't specify curly braces for one-line conditions.